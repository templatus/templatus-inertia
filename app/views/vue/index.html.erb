<div id='vue-app'></div>
