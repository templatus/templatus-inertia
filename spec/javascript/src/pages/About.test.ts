import { mount } from '@vue/test-utils';
import AboutPage from '@/pages/AboutPage.vue';
import * as Vue from 'vue';

describe('About', () => {
  const savedVersion = Object.getOwnPropertyDescriptor(
    Vue,
    'version',
  ) as PropertyDescriptor;

  beforeAll(() => {
    // Mock `Vue.version` to have a stable snapshot
    Object.defineProperty(Vue, 'version', {
      value: '3.x.y',
    });
  });

  afterAll(() => {
    Object.defineProperty(Vue, 'version', savedVersion);
  });

  test('matches snapshot', () => {
    const wrapper = mount(AboutPage, {});

    expect(wrapper.html()).toMatchSnapshot();
  });
});
