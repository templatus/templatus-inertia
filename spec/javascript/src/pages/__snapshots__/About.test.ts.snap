// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
<h1 class="break-words text-4xl font-extrabold tracking-tight text-rails-dark sm:text-5xl lg:text-6xl">About</h1>
<p class="mt-6 max-w-4xl text-base lg:text-xl"> Templatus is an opinionated template for web applications with <b>Ruby on Rails</b> and <b>Vue.js 3</b></p>
<div class="mt-20 space-y-24">
  <div>
    <div class="lg:grid lg:grid-cols-3 lg:gap-x-8">
      <div>
        <h2 class="text-base font-semibold uppercase tracking-wide text-rails-dark">Server side</h2>
        <p class="mt-2 text-3xl font-extrabold text-gray-900">The backend</p>
      </div>
      <div class="mt-12 lg:col-span-2 lg:mt-0">
        <dl class="space-y-10 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-10 sm:space-y-0 lg:gap-x-8">
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://www.ruby-lang.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">Ruby</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">A dynamic, open source programming language with a focus on simplicity and productivity. It has an elegant syntax that is natural to read and easy to write.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://rubyonrails.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">Ruby on Rails</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Rails is a web application development framework written in the Ruby programming language. It is designed to make programming web applications easier by making assumptions about what every developer needs to get started.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://www.postgresql.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">PostgreSQL</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">PostgreSQL is a powerful, open source object-relational database system with over 30 years of active development that has earned it a strong reputation for reliability, feature robustness, and performance.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://puma.io/" class="hover:underline" target="_blank" rel="noopener noreferrer">Puma</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Puma is a simple, fast, multi-threaded, and highly concurrent HTTP 1.1 server for Ruby/Rack applications.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://redis.io/" class="hover:underline" target="_blank" rel="noopener noreferrer">Redis</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Redis is an open source, in-memory data structure store, used as a database, cache, and message broker.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://sidekiq.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">Sidekiq</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Simple, efficient background processing for Ruby<p class="mt-2 text-xs">→ <a class="font-medium uppercase hover:underline" href="/sidekiq">Admin frontend</a></p>
            </dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
  <div>
    <div class="lg:grid lg:grid-cols-3 lg:gap-x-8">
      <div>
        <h2 class="text-base font-semibold uppercase tracking-wide text-rails-dark">Client side</h2>
        <p class="mt-2 text-3xl font-extrabold text-gray-900">The frontend</p>
      </div>
      <div class="mt-12 lg:col-span-2 lg:mt-0">
        <dl class="space-y-10 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-10 sm:space-y-0 lg:gap-x-8">
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://v3.vuejs.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">Vue</a><span class="ml-2 inline-flex items-center rounded-full bg-rose py-0.5 px-2.5 text-xs font-medium text-black">3.x.y</span></p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Vue is a progressive framework for building user interfaces.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://tailwindcss.com/" class="hover:underline" target="_blank" rel="noopener noreferrer">Tailwind CSS 3</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">A utility-first CSS framework packed with classes that can be composed to build any design, directly in your markup.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://www.typescriptlang.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">TypeScript</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">TypeScript is an open-source language which builds on JavaScript, one of the world’s most used tools, by adding static type definitions.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://pinia.esm.dev/" class="hover:underline" target="_blank" rel="noopener noreferrer">Pinia</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Intuitive, type safe, light and flexible Store for Vue using the composition api with DevTools support</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://headlessui.dev/" class="hover:underline" target="_blank" rel="noopener noreferrer">HeadlessUI</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Completely unstyled, fully accessible UI components, designed to integrate beautifully with Tailwind CSS.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://heroicons.com/" class="hover:underline" target="_blank" rel="noopener noreferrer">Heroicons</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Beautiful hand-crafted SVG icons, by the makers of Tailwind CSS.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://vitejs.dev/" class="hover:underline" target="_blank" rel="noopener noreferrer">Vite</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Next generation frontend tooling. It's fast!</dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
  <div>
    <div class="lg:grid lg:grid-cols-3 lg:gap-x-8">
      <div>
        <h2 class="text-base font-semibold uppercase tracking-wide text-rails-dark">Quality insurance</h2>
        <p class="mt-2 text-3xl font-extrabold text-gray-900">Continuous integration</p>
      </div>
      <div class="mt-12 lg:col-span-2 lg:mt-0">
        <dl class="space-y-10 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-10 sm:space-y-0 lg:gap-x-8">
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://github.com/features/actions" class="hover:underline" target="_blank" rel="noopener noreferrer">GitHub Actions</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Test, build and deliver on every push to GitHub</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://rspec.info/" class="hover:underline" target="_blank" rel="noopener noreferrer">RSpec</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Behaviour Driven Development for Ruby. Making TDD Productive and Fun.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://rubocop.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">RuboCop</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">The Ruby Linter/Formatter that Serves and Protects</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://jestjs.io/" class="hover:underline" target="_blank" rel="noopener noreferrer">Jest</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Jest is a delightful JavaScript Testing Framework with a focus on simplicity</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://eslint.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">ESLint</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">ESLint statically analyzes your code to quickly find problems</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://cypress.io/" class="hover:underline" target="_blank" rel="noopener noreferrer">Cypress</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Fast, easy and reliable testing for anything that runs in a browser.</dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
  <div>
    <div class="lg:grid lg:grid-cols-3 lg:gap-x-8">
      <div>
        <h2 class="text-base font-semibold uppercase tracking-wide text-rails-dark">Ship it!</h2>
        <p class="mt-2 text-3xl font-extrabold text-gray-900">Deployment</p>
      </div>
      <div class="mt-12 lg:col-span-2 lg:mt-0">
        <dl class="space-y-10 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-10 sm:space-y-0 lg:gap-x-8">
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://alpinelinux.org/" class="hover:underline" target="_blank" rel="noopener noreferrer">Alpine Linux</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Alpine Linux is an independent, non-commercial, general purpose Linux distribution designed for power users who appreciate security, simplicity and resource efficiency.</dd>
          </div>
          <div class="relative">
            <dt><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="absolute h-6 w-6 text-vue">
                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7"></path>
              </svg>
              <p class="ml-9 flex items-center text-lg font-medium leading-6 text-rails-dark"><a href="https://www.docker.com/" class="hover:underline" target="_blank" rel="noopener noreferrer">Docker</a>
                <!--v-if-->
              </p>
            </dt>
            <dd class="mt-2 ml-9 text-base text-gray-600">Dockerized using an optimized base image.</dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
